kairos-server/src/lib.rs

```diff
@@ -22,3 +25,15 @@ pub fn app_router(state: Arc<state::BatchStateManager>) -> Router {
         .typed_post(routes::transfer_handler)
         .with_state(state)
 }
+
+pub async fn run(config: ServerConfig) {
+    tracing_subscriber::fmt::init();
+
+    let app = app_router(BatchStateManager::new_empty());
+
+    tracing::info!("starting http server on `{}`", config.socket_addr);
+    let listener = tokio::net::TcpListener::bind(config.socket_addr)
+        .await
+        .unwrap();
```

Comment:
**Suggestion:** Since we allow binding to port 0, it would be useful to log the address with bound port:

```suggestion
    let listener = tokio::net::TcpListener::bind(config.socket_addr)
        .await
        .unwrap();
    tracing::info!("listening on `{}`", listener.local_addr().unwrap());
```

Example logs for `KAIROS_SERVER_SOCKET_ADDR=0.0.0.0:0`:

```
2024-05-06T11:05:38.197541Z  INFO kairos_server: starting http server on `0.0.0.0:0`
2024-05-06T11:05:38.197591Z  INFO kairos_server: listening on `0.0.0.0:43653`
```